aya-common/src/lib.rs

```diff
@@ -0,0 +1,296 @@
+                spec.arg_type = UsdtArgType::Const;
+                spec.val_off = caps.get(6).unwrap().as_str().parse::<i64>().map_err(|e| {
+                    ParseError::UsdtArgSpecError(format!("could not parse {}: {}", s, e))
+                })? as u64;
+            }
+            if arg_size < 0 {
+                spec.arg_signed = true;
+                arg_size = -arg_size;
+            }
+            match arg_size {
+                1 | 2 | 4 | 8 => spec.arg_bitshift = (arg_size * 8) as i8,
+                _ => {
+                    return Err(ParseError::UsdtArgSpecError(format!(
+                        "arg size was not 1,2,4,8: {}",
+                        s
+                    )))
+                }
+            }
+            Ok(spec)
+        }
+    }
+
+    #[cfg(target_arch = "x86_64")]
+    fn calc_pt_regs_offset(reg: &str) -> Result<i16, ParseError> {
+        use crate::generated::pt_regs;
+        use memoffset::offset_of;
+        match reg {
+            "rip" | "eip" => Ok(offset_of!(pt_regs, rip) as i16),
+            "rax" | "eax" | "ax" | "al" => Ok(offset_of!(pt_regs, rax) as i16),
+            "rbx" | "ebx" | "bx" | "bl" => Ok(offset_of!(pt_regs, rbx) as i16),
+            "rcx" | "ecx" | "cx" | "cl" => Ok(offset_of!(pt_regs, rcx) as i16),
+            "rdx" | "edx" | "dx" | "dl" => Ok(offset_of!(pt_regs, rdx) as i16),
+            "rsi" | "esi" | "si" | "sil" => Ok(offset_of!(pt_regs, rsi) as i16),
+            "rdi" | "edi" | "di" | "dil" => Ok(offset_of!(pt_regs, rdi) as i16),
+            "rbp" | "ebp" | "bp" | "bpl" => Ok(offset_of!(pt_regs, rbp) as i16),
+            "rsp" | "esp" | "sp" | "bsl" => Ok(offset_of!(pt_regs, rsp) as i16),
+            "r8" | "r8d" | "r8w" | "r8b" => Ok(offset_of!(pt_regs, r8) as i16),
+            "r9" | "r9d" | "r9w" | "r9b" => Ok(offset_of!(pt_regs, r9) as i16),
+            "r10" | "r10d" | "r10w" | "r10b" => Ok(offset_of!(pt_regs, r10) as i16),
+            "r11" | "r11d" | "r11w" | "r11b" => Ok(offset_of!(pt_regs, r11) as i16),
+            "r12" | "r12d" | "r12w" | "r12b" => Ok(offset_of!(pt_regs, r12) as i16),
+            "r13" | "r13d" | "r13w" | "r13b" => Ok(offset_of!(pt_regs, r13) as i16),
+            "r14" | "r14d" | "r14w" | "r14b" => Ok(offset_of!(pt_regs, r14) as i16),
+            "r15" | "r15d" | "r15w" | "r15b" => Ok(offset_of!(pt_regs, r15) as i16),
+            _ => Err(ParseError::UsdtArgSpecError(format!(
+                "unknown register: {}",
+                reg
+            ))),
+        }
+    }
+
+    #[cfg(target_arch = "aarch64")]
+    fn calc_pt_regs_offset(reg: &str) -> Result<i16, ParseError> {
+        use crate::generated::user_pt_regs;
+        use memoffset::offset_of;
+        use std::mem;
+        match reg {
+            r if r.starts_with('x') => {
+                let n: usize = r.strip_prefix('x').unwrap().parse().unwrap();
+                Ok((offset_of!(user_pt_regs, regs) + (mem::size_of::<u64>() * n)) as i16)
```

Comment:
If the only difference between `aarch64` and `arm` implementation is going to be `u64` vs `c_long`, maybe we could keep the common code as a macro to avoid code duplication?

```rust
macro_rules! calc_pt_regs_offset_arm {
    ($int_ty:ty) => {
        use crate::generated::pt_regs;
        use memoffset::offset_of;
        use std::mem;
        match reg {
            // TODO: This assumes the notation is the same as aarch64
            // This needs testing and potentially updating
            r if r.starts_with('x') => {
                let n: usize = r.strip_prefix('x').unwrap().parse().unwrap();
                Ok(
                    (offset_of!(pt_regs, uregs) + (mem::size_of::<$int_ty>() * n))
                        as i16,
                )
            }
            _ => Err(ParseError::UsdtArgSpecError(format!(
                "unknown register: {}",
                reg
            ))),
        }
    }
}
```

Reply:
It's not the only difference. aarch64 has additional registers like "sp" and "pc" whereas the armv7 doesnt.
also it's likely the terms used to describe those regs are different between aarch64/armv7 so it's best to keep these separate.